icu4c/source/common/rbbitblb.cpp

```diff
@@ -1082,21 +1082,21 @@ bool RBBITableBuilder::findDuplCharClassFrom(IntPair *categories) {
     int32_t numStates = fDStates->size();
     int32_t numCols = fRB->fSetBuilder->getNumCharCategories();
 
-    uint16_t table_base;
-    uint16_t table_dupl;
     for (; categories->first < numCols-1; categories->first++) {
         for (categories->second=categories->first+1; categories->second < numCols; categories->second++) {
-             for (int32_t state=0; state<numStates; state++) {
-                 RBBIStateDescriptor *sd = (RBBIStateDescriptor *)fDStates->elementAt(state);
-                 table_base = (uint16_t)sd->fDtran->elementAti(categories->first);
-                 table_dupl = (uint16_t)sd->fDtran->elementAti(categories->second);
-                 if (table_base != table_dupl) {
-                     break;
-                 }
-             }
-             if (table_base == table_dupl) {
-                 return true;
-             }
+            uint16_t table_base = 0;
+            uint16_t table_dupl = 1;
```

Comment:
I think you moved these two variables down, and gave them "default" values because the compiler warning was that if `numStates` is zero, then the `if` statement below might be comparing uninitialized memory right?
But it seems you gave them *different* values so as to prevent it returning `true` (that it contains duplicates) in the case with a zero size.  I think it might be worth adding a comment about the reason why.

Reply:
That's correct, I'll add a comment explaining that.